hathor/transaction/resources/mempool.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from typing import TYPE_CHECKING
+
+from twisted.web import resource
+
+from hathor.api_util import set_cors
+from hathor.cli.openapi_files.register import register_resource
+
+if TYPE_CHECKING:
+    from twisted.web.http import Request
+
+    from hathor.manager import HathorManager
+
+
+@register_resource
+class MempoolResource(resource.Resource):
+    """ Implements a web server API to return transactions on the mempool.
+
+    You must run with option `--status <PORT>`.
+    """
+    isLeaf = True
+
+    def __init__(self, manager: 'HathorManager'):
+        # Important to have the manager so we can know the tx_storage
+        self.manager = manager
+
+    def render_GET(self, request: 'Request') -> bytes:
+        """ Get request /mempool/ that returns transactions on the mempool
+
+            :rtype: string (json list)
+        """
+        request.setHeader(b'content-type', b'application/json; charset=utf-8')
+        set_cors(request, 'GET')
+
```

Comment:
We should add a limit here. We know the mempool is not supposed to be huge but we can already add this protection. Maybe limit to 100? In this case we can return in the response also `truncated: True`. And leave a TODO comment to add pagination in the future.

And I am not sure only the tx_id is enough for some cases, maybe we can have a parameter to return the full tx?

Reply:
I'm putting a limit of 100 transactions for now, but i don't think the `truncated: True` is necessary, since the caller won't be able to fetch the rest of the mempool (in this version).

Pagination is a bit tricky, i had a conversation with @jansegre and @msbrogli about this and i don't think the intention is to add this feature to the index (for performance reasons).

I don't think we should change the "type" of the response based on an argument. If we need to send the full tx we should do it here, or add another api with this behavior. What do you think?

Reply:
I understand that they won't be able to fetch the rest of the mempool but maybe it's useful for them to know if they are not getting the full mempool.

About the pagination I agree to leave without it. And to return only one type.

Reply:
I'm fine with this as is. Something has to go really wrong for the mempool to grow too large, and iterating it is really fast (we don't do any processing, it should be `O(N)` for `N` txs on the mempool).